tests/jenkins/TestMessages.groovy

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
+import static com.lesfurets.jenkins.unit.global.lib.ProjectSource.projectSource
+import org.junit.Before
+import org.junit.Test
+import com.lesfurets.jenkins.unit.declarative.DeclarativePipelineTest
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import com.lesfurets.jenkins.unit.MethodCall
+import static org.junit.Assert.*
+import static org.assertj.core.api.Assertions.assertThat
+import java.util.*
+
+class TestMessages extends DeclarativePipelineTest {
+
+    String sharedLibs = ''
+
+    @Override
+    @Before
+    void setUp() throws Exception {
+        scriptRoots += 'tests/jenkins/jobs'   
+
+        super.setUp()
+
+        helper.registerAllowedMethod('findFiles', [Map.class], null)
+        
+        helper.registerSharedLibrary(
+            library().name('jenkins')
+                .defaultVersion('<notNeeded>')
+                .allowOverride(true)
+                .implicit(true)
+                .targetPath('<notNeeded>')
+                .retriever(projectSource())
+                .build()
+            )
+    }
+
+    @Test
+    void testMessages() throws Exception {
+        def script = runScript("messages.groovy")
+
+        assertArrayEquals(Arrays.asList(
+            "{includes=messages/*, name=messages-stage1}",
+            "{includes=messages/*, name=messages-stage2}"
+        ).toArray(), helper.callStack.stream()
+            .filter { c -> c.methodName == "stash" }
+            .map(MethodCall.&callArgsToString)
+            .collect()
+            .toArray()
+        )
```

Comment:
Could you explain to me why these 2 arrays are equal after all despite different input?
I am not very familiar with JUnit tests.

Thanks.

Reply:
These are the parameters to all the `stash` calls made in the workflow via `Messages.add`. The first stashes all messages into `messages-stage1`, the second into `messages-stage2`. The first argument here is what we expect (parameters to those two calls) and the second is all the `stash` calls from the call stack of the Jenkins workflow (`.filter { c -> c.methodName == "stash" }`), all the arguments from those calls (`map(MethodCall.&callArgsToString)`), transformed into an array.

Reply:
Got it. Thanks.